### DISCUSSION TITLE
child-SA 1 is not trying to connect after deleting child-sa 2

### DISCUSSION THREAD

**Discussion by amdrsantos** [strongswan/strongswan#968] [Q&A]
We have the following scenario:
    We have a machine A and machine B. Where machine B is fully configured and machine A is not configured.
    1) Machine A is not able to communicate IKE packets with B.
    2) In machine A, we add a new IKE configuration using vici, load-conn
        ike-server{
            (...)
            children{
                child-1{
                    (...)
                    dpd-action = start
                }
            }
        }

        child-1 initiates

    3) In machine A, we add a new child in the same IKE using vici, load-conn
        ike-server{
            (...)
            children{
                child-1{
                    (...)
                    dpd-action = start
                }
                child-2{
                    (...)
                    dpd-action = start
                }
            }
        }
        child-2 initiates

    4) In machine A, we remove child-2 from the configuration using vici, load-conn
        ike-server{
            (...)
            children{
                child-1{
                    (...)
                    dpd-action = start
                }
            }
        }

        **IKE_SA is deleted, even though child-1 was trying to initiate.**

    5) Recover connection between machines A and B
        Because IKE_SA was deleted, Machine A no longer initiates a connection with machine B.

Is this normal behavior? I was expecting that child-1 kept trying to connect.

Thanks, Alex

**Comment by tobiasbrunner** [comment]
Note that `dpd_action` is unrelated to what's happening when connections are unloaded, the relevant option is `start_action`.

> Is this normal behavior?

Currently it is, because queued child-create tasks are not considered when deciding if the complete IKE_SA should be destroyed, only established CHILD_SAs. You might want to consider using `start_action=trap` so outbound traffic triggers initiating the CHILD_SA.

**Reply by amdrsantos** [reply]
Hi Tobias,

Thank you for your feedback.
But if I configure both ends with start_action=trap, they will not start connecting automatically.
Both will wait for each other indefinitely.

Regards, Alex

**Reply by tobiasbrunner** [reply]
The connection will be started on-demand once there is matching traffic to protect.
